packages/core/src/components/validation/validation.ts

```diff
@@ -1,3 +1,4 @@
 export interface ValidationProps {
+  for?: keyof ValidityState;
   state: 'error';
```

Comment:
[No action required] This now seems little odd - `for` is validity state, but then there's another `state` that's an error. Maybe prop should be named differently, like `variant`.

Reply:
Well the idea is for the API to read like `<Validation state="error" for="valueMissing" />`, the `ValidityState` is just the object that holds which validation error an element has.

We could change the name for sure, `variant` does not convey the intent though.

Maybe `when`? Or `if`?

Reply:
Oh boy, that's a breaking change. I see your point though.

In hindsight `variant` or `kind` would be better for sure, aligns with Button.

Reply:
I don't mind either, you pick.

For the `state` prop - I agree this would be a breaking change.

Let's come back to that one when we later have to add any other "state", and then probably will need to do a breaking change...

Reply:
Cool, I'll go with `if` then.

I don't know we'll necessarily need a breaking change, presumably adding other `state`s like `success` would just be a `feat`.

But yea let's think about it then.
We could support both for a while and deprecate `state`.
Like
```ts
kind: 'error' | 'success';
@deprecated Use `kind` instead.
state: 'error';
```

And then just remove `state` on 2.0.0.

Reply:
Sure, should this be done _now_ then (in a separate PR)?